packages/backend/src/tools/uif/multi/types.ts

```diff
@@ -31,3 +31,8 @@ export interface ConfigurationRange<T> {
   to: number
   configurations: Configuration<T>[]
 }
+
+export interface DatabaseMiddleware {
```

Comment:
Let's hide implementation details and alias the type of database transaction. Basically all classes using `DatabaseMiddleware` should refer to `DatabaseTransaction` rather than `Knex.Transaction`

```
export type DatabaseTransaction = Knex.Transaction

export interface DatabaseMiddleware {
  add: (cb: (trx?: DatabaseTransaction) => Promise<void>) => void
  execute: () => Promise<void>
}
```

I would also move `DatabaseMiddleware` and `KnexMiddleware` to some more generic place like `src/peripherals/database`